packages/components/src/draggable/index.native.js

```diff
@@ -37,37 +41,38 @@ export default function Draggable( {
 	const isAnyTextInputFocused = useSharedValue( false );
 
 	const checkTextInputFocus = () => {
-		isAnyTextInputFocused.value =
+		const isTextInputFocused =
 			TextInputState.currentlyFocusedInput() !== null;
+		isAnyTextInputFocused.value = isTextInputFocused;
+		return isTextInputFocused;
+	};
+
+	const dragStart = ( event ) => {
+		const isTextInputFocused = checkTextInputFocus();
+
+		if ( ! isTextInputFocused && onDragStart ) {
+			runOnUI( onDragStart )( event );
+		}
+	};
+
+	const dragEnd = () => {
+		const isTextInputFocused = checkTextInputFocus();
+
+		if ( ! isTextInputFocused && onDragEnd ) {
+			runOnUI( onDragEnd )();
+		}
```

Comment:
I noticed while testing on Android, that we might have a race condition when updating the `isAnyTextInputFocused` shared value, between the `onBegin` and `onStart` handlers of the `LongPress` gesture. In some, cases I've seen that the shared value was `false` when triggering `onStart` even though a text input was focused. In order to prevent this, I updated the drag events handlers so now we always check if any text input is focused before triggering them.